pkg/rlptools/wasm_utils.go

```diff
@@ -245,7 +245,7 @@ func dataFromLimt(limitIdentifier string, limit *kuadrantv1beta2.Limit) (data []
 	data = append(data, wasm.DataItem{Static: &wasm.StaticSpec{Key: limitIdentifier, Value: "1"}})
 
 	for _, counter := range limit.Counters {
-		data = append(data, wasm.DataItem{Selector: &wasm.SelectorSpec{Selector: counter}})
+		data = append(data, wasm.DataItem{Selector: &wasm.SelectorSpec{Selector: counter, Default: common.Ptr("")}})
```

Comment:
What if I set as counter `request.headers.My-Custom-Header-01` ?  All the requests not having that header will contribute to a single (limitador's internal hit counter)

Reply:
Request that do not have that header are treated as requests that have that header with an empty value, therefore sharing the same counter. If what you want is to trigger or not trigger a limit based on the presence of a header, use a condition instead; not a counter qualifier.

Reply:
Indeed, what you say is correct, but I do not fully agree on this UX: if you set a counter based on a header, add a condition on the header existence. 

For me, adding a counter implicitly tells I **only** want to rate limit traffic that is qualified by the counter. 

Let's see what @didierofrivia and @alexsnaps have to say about this

Reply:
@eguzki, coincidently, this was exactly the example that made us[^1] want to default to an empty string in the wasm-shim (what https://github.com/Kuadrant/wasm-shim/pull/39 wants to revert now).

[^1]: Discussed offline with @alexsnaps and @didierofrivia on 2023-07-27, and [implemented](https://github.com/Kuadrant/wasm-shim/pull/37/commits/59033b0b4f89290391c8eee9acf00fde51fe6654) by @alexsnaps on 2023-07-28.

Reply:
https://github.com/Kuadrant/wasm-shim/pull/39 does not revert entirely [that commit](https://github.com/Kuadrant/wasm-shim/pull/37/commits/59033b0b4f89290391c8eee9acf00fde51fe6654). Only reverts the behavior when the descriptor is not found in the module generating RLS descriptors. For the conditions, https://github.com/Kuadrant/wasm-shim/pull/39 keeps the changes to allow operators to be meaningful.

Reply:
> [Kuadrant/wasm-shim#39](https://github.com/Kuadrant/wasm-shim/pull/39) does not revert entirely [that commit](https://github.com/Kuadrant/wasm-shim/pull/37/commits/59033b0b4f89290391c8eee9acf00fde51fe6654).

I understand that. But, most importantly, I also understand that setting the default value to `""`, so some RL is enforced, is one approach. In itself, it is nor better, nor worse than setting no default at all. If most people prefers the contrary of what's proposed here, I'm good with that too obviously.